crescent/plugin.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import importlib
+from typing import TypeVar, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .bot import Bot
+    from .internal.meta_struct import MetaStruct
+
+_T = TypeVar("_T", bound="MetaStruct")
```

Comment:
This can be moved into the `if TYPE_CHECKING` block along with the `TypeVar` import. Also can you rename it to `T` because I've done all my generics with an underscore.